dashboard/app/views/layouts/_header.html.haml

```diff
@@ -106,7 +106,7 @@
 
     script_data = script.summarize_header
     lesson_group_data = script.lesson_groups.map(&:summarize)
-    stage_data = (@stage || script_level.lesson).summarize
+    stage_data = (@stage || script_level.lesson).summarize(true)
```

Comment:
Why is this change needed?

Also, very minor, including the param name here might help readability.

Reply:
that `true` is for the `include_bonus_levels` param, but i couldn't figure out how to include the param name. i figured it out now, and imo `summarize(include_bonus_levels = true)` explains the need for the change, but lmk if you think i should also add a comment.

Reply:
Sorry, I'm still not clear what's going on here, probably because I don't have a clear picture of where `stage_data` ends up getting used.  Where does the information about the bonus levels eventually get used?  What happened before when we didn't have this information?

Reply:
that data is passed down to the `LessonProgress` component used in the header. previously, since we didn't indicate user progress for bonus levels in the header, we didn't need any level-specific data. the flag was conditionally displayed based on the presence of `lessonExtrasUrl` (still true), and it was colored green if `isLessonExtras` (now that value is only used to determine if lesson extras is "selected"). to enable the flag to be colored based on user progress, we now need to include the level-specific data for the bonus levels. this is handled in `LessonProgress.isOnBonusLevel()` and `LessonProgress.isBonusComplete()`